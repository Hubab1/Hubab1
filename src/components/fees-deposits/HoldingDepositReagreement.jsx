import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { APPLICATION_EVENTS, ROUTES } from 'app/constants';
import captureRoute from 'app/captureRoute';
import { PaymentTerms } from 'components/fees-deposits/PaymentTerms';
import API from 'app/api';
import { fetchApplicant } from 'reducers/applicant';
import { selectors } from 'reducers/renter-profile';

export const HoldingDepositReagreement = ({ profile, configuration, initialPage, history, _next }) => {
    if (!profile || !configuration || !profile.lease_start_date) {
        return null;
    }

    const unitNumber = profile.unit?.unit_number;
    const communityName = profile.community?.display_name;
    const leaseStartDate = profile.lease_start_date;
    const holdingDepositAmount =
        parseFloat(profile.selected_rental_options?.['holding-deposit']?.[0]?.quoted_deposit_amount) || 0;

    const handleTermsAccepted = (data) => {
        API.acceptTerms(data).then(async () => {
            const newApplicant = await API.fetchApplicant();
            const needToReagree = newApplicant.events.findIndex(
                (e) => parseInt(e.event) === parseInt(APPLICATION_EVENTS.MILESTONE_APPLICANT_NEEDS_TO_REAGREE_TO_HD)
            );
            console.log(needToReagree);
            if (needToReagree !== -1) {
                newApplicant.events.splice(needToReagree, 1);
            }

            console.log(initialPage);
            console.log(_next);
            history.push(initialPage);
        });
    };

    return (
        <PaymentTerms
            holdingDepositAmount={holdingDepositAmount}
            unitNumber={unitNumber}
            communityName={communityName}
            leaseStartDate={leaseStartDate}
            canProceedToPayment={true}
            isReagreeing={true}
            reagree={(data) => handleTermsAccepted(data)}
        />
    );
};

HoldingDepositReagreement.propTypes = {
    profile: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
    initialPage: PropTypes.string,
    history: PropTypes.object,
    _next: PropTypes.string,
    fetchApplicant: PropTypes.func,
};

const mapStateToProps = (state) => ({
    profile: state.renterProfile,
    configuration: state.configuration,
    initialPage: selectors.selectDefaultInitialPage(state),
    _next: selectors.selectNextRoute(state),
});

const mapDispatchToProps = {
    fetchApplicant,
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(captureRoute(HoldingDepositReagreement, ROUTES.HOLDING_DEPOSIT_TERMS_AGREEMENT));
