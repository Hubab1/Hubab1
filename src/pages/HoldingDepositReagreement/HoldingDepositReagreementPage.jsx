import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'constants/constants';
import API from 'api/api';
import captureRoute from 'utils/captureRoute';

import PaymentTerms from 'common-components/PaymentTerms/PaymentTerms';

export const HoldingDepositReagreementPage = ({ profile, configuration }) => {
    if (!profile || !configuration || !profile.lease_start_date) {
        return null;
    }

    const unitNumber = profile.unit?.unit_number;
    const communityName = profile.community?.display_name;
    const leaseStartDate = profile.lease_start_date;
    const holdingDepositAmount =
        parseFloat(profile.selected_rental_options?.['holding-deposit']?.[0]?.quoted_deposit_amount) || 0;

    const handleTermsAccepted = (data) => {
        API.acceptTerms(profile.id, data).then(async () => {
            window.location.reload();
        });
    };

    return (
        <PaymentTerms
            holdingDepositAmount={holdingDepositAmount}
            unitNumber={unitNumber}
            communityName={communityName}
            leaseStartDate={leaseStartDate}
            canProceedToPayment={true}
            isReagreeing={true}
            handleTermsAccepted={(data) => handleTermsAccepted(data)}
            application={profile}
        />
    );
};

HoldingDepositReagreementPage.propTypes = {
    profile: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
    profile: state.renterProfile,
    configuration: state.configuration,
});

export default connect(mapStateToProps)(
    captureRoute(HoldingDepositReagreementPage, ROUTES.HOLDING_DEPOSIT_TERMS_AGREEMENT)
);
