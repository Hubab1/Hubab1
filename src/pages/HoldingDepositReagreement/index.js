import HoldingDepositReagreementPage from './HoldingDepositReagreementPage';

export default HoldingDepositReagreementPage;
