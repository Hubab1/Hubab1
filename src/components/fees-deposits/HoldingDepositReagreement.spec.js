import React from 'react';
import { shallow } from 'enzyme';
import { HoldingDepositReagreement } from './HoldingDepositReagreement';
import API from 'app/api';
import mockProfile from 'reducers/mock-profile.json';
import mockConfig from 'reducers/mock-config.json';

let defaultProps;
beforeEach(() => {
    defaultProps = {
        profile: mockProfile,
        configuration: mockConfig,
    };
    API.fetchHoldingDepositTerms = jest.fn().mockResolvedValue('');
});

it('Matches snapshot without a start date', () => {
    const wrapper = shallow(<HoldingDepositReagreement {...defaultProps} />);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Matches snapshot with a start date', () => {
    defaultProps.profile.lease_start_date = '14/22/2050';
    const wrapper = shallow(<HoldingDepositReagreement {...defaultProps} />);
    expect(wrapper.getElement()).toMatchSnapshot();
});
