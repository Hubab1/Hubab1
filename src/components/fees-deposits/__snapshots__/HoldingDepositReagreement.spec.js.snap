// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot with a start date 1`] = `
<PaymentTerms
  canProceedToPayment={true}
  holdingDepositAmount={501.5}
  isReagreeing={true}
  leaseStartDate="14/22/2050"
  reagree={[Function]}
  unitNumber="12a"
/>
`;

exports[`Matches snapshot without a start date 1`] = `null`;
